src/django/api/tiler.py

```diff
@@ -0,0 +1,62 @@
+from django.contrib.gis.geos import Polygon
+from django.db import connection
+from mercantile import bounds
+
+from api.querysets import get_facility_queryset_from_query_params
+
+
+def get_facilities_vector_tile(params, layer, z, x, y):
+    """
+    Create a vector tile for a layer generated via PostGIS's `ST_AsMVT`
+    function, filtered by params.
+
+    Arguments:
+    params (dict) -- Request query parameters whose potential choices are
+                     enumerated in `api.constants.FacilitiesQueryParams`
+    layer (string) -- The name of the tile layer. Currently only "facilities"
+                      is supported.
+    z (int) -- Zoom level.
+    x (int) -- X (horizontal) position for requested tile on a grid.
+    y (int) -- Y (vertical) position for requested tile on a grid.
+
+    Returns:
+    A vector tile.
+    """
+    tile_bounds = bounds(x, y, z)
+
+    mvt_geom_query = """
+        ST_AsMVTGeom(
+            location,
+            ST_MakeEnvelope({xmin}, {ymin}, {xmax}, {ymax}, 4326),
+            4096,
+            1000,
+            true
+        ) """
+
+    filter_polygon = Polygon.from_bbox((
+        tile_bounds.west, tile_bounds.south,
+        tile_bounds.east, tile_bounds.north)).buffer(1000)
```

Comment:
Our filter polygon is in lat/lng which has min/max coordinate values between -90/90 and -180/180. Using 1000 as a buffer amount creates a polygon that covers the entire world.

<img width="1120" alt="Screen Shot 2019-08-20 at 9 20 09 AM" src="https://user-images.githubusercontent.com/17363/63371229-8c3e7880-c338-11e9-8d13-4a1c27678fa1.png">

Consider this alternate logic that creates a relative buffer based on a percentage of the real-world tile size. Lat and lng have different scales so we need to calculate different buffers for the individual axes.

```diff
diff --git a/src/django/api/tiler.py b/src/django/api/tiler.py
index df05397..c643e2e 100644
--- a/src/django/api/tiler.py
+++ b/src/django/api/tiler.py
@@ -33,9 +33,15 @@ def get_facilities_vector_tile(params, layer, z, x, y):
             true
         ) """
 
+    filter_buffer_percent = 0.2
+    ew_buffer = \
+        abs(tile_bounds.east - tile_bounds.west) * filter_buffer_percent
+    ns_buffer = \
+        abs(tile_bounds.north - tile_bounds.south) * filter_buffer_percent
+
     filter_polygon = Polygon.from_bbox((
-        tile_bounds.west, tile_bounds.south,
-        tile_bounds.east, tile_bounds.north)).buffer(1000)
+        tile_bounds.west - ew_buffer, tile_bounds.south - ns_buffer,
+        tile_bounds.east + ew_buffer, tile_bounds.north + ns_buffer))
 
     query, params_for_sql = get_facility_queryset_from_query_params(params) \
         .filter(location__within=filter_polygon) \
```

Reply:
Changed this in 732acef0f71ec560f2854c6150a1f7e3a207506a